Source/WebCore/dom/Subscriber.cpp

```diff
@@ -0,0 +1,162 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "config.h"
+#include "Subscriber.h"
+
+#include "AbortController.h"
+#include "AbortSignal.h"
+#include "SubscriberCallback.h"
+#include "SubscriptionObserverCallback.h"
+#include "VoidCallback.h"
+
+namespace WebCore {
+
+Ref<Subscriber> Subscriber::create(ScriptExecutionContext& context)
+{
+    return adoptRef(*new Subscriber(context, nullptr, nullptr, nullptr));
+}
+
+Ref<Subscriber> Subscriber::create(ScriptExecutionContext& context, RefPtr<SubscriptionObserverCallback> nextCallback)
+{
+    return adoptRef(*new Subscriber(context, nextCallback, nullptr, nullptr));
+}
+
+Ref<Subscriber> Subscriber::create(ScriptExecutionContext& context, RefPtr<SubscriptionObserverCallback> nextCallback, RefPtr<SubscriptionObserverCallback> errorCallback, RefPtr<VoidCallback> completeCallback)
+{
+    return adoptRef(*new Subscriber(context, nextCallback, errorCallback, completeCallback));
+}
+
+Subscriber::Subscriber(ScriptExecutionContext& context, RefPtr<SubscriptionObserverCallback> nextCallback, RefPtr<SubscriptionObserverCallback> errorCallback, RefPtr<VoidCallback> completeCallback)
+    : ActiveDOMObject(&context)
+    , m_abortController(AbortController::create(context))
+    , m_next(WTFMove(nextCallback))
+    , m_error(WTFMove(errorCallback))
+    , m_complete(WTFMove(completeCallback))
+{
+    followSignal(m_abortController->signal());
+    suspendIfNeeded();
+}
+
+void Subscriber::next(JSC::JSValue value)
+{
+    if (!isActive())
+        return;
+
+    if (m_next)
+        m_next->handleEvent(value);
+}
+
+void Subscriber::error(JSC::JSValue value)
+{
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    if (!m_active) {
+        reportException(
+            JSC::jsCast<JSDOMGlobalObject*>(context->globalObject()),
+            JSC::Exception::create(context->vm(), value));
+        return;
+    }
+
+    if (!isActive())
+        return;
+
+    auto error = m_error;
+
+    close(JSC::jsUndefined());
+
+    if (error)
+        error->handleEvent(value);
+
+    else
+        reportException(
+            JSC::jsCast<JSDOMGlobalObject*>(context->globalObject()),
+            JSC::Exception::create(context->vm(), value));
+}
+
+void Subscriber::complete()
+{
+    if (!isActive())
+        return;
+
+    auto complete = m_complete;
+
+    close(JSC::jsUndefined());
+
+    if (complete)
+        complete->handleEvent();
+}
+
+void Subscriber::addTeardown(Ref<VoidCallback> callback)
+{
+    if (m_active && !isActive())
+        return;
+
+    if (m_active)
+        m_teardowns.add(callback);
+    else
+        callback.get().handleEvent();
```

Comment:
Instead of using `get` whenever you're dereferencing a `Ref` or a `RefPtr` you can use `operator->`. So:

```C++
callback->handleEvent()
```